packages/mail-client/src/mail-client.spec.ts

```diff
@@ -44,6 +56,47 @@ describe('mail client', () => {
     expect(spySendMail).toBeCalledTimes(2);
     expect(spySendMail).toBeCalledWith(mailOptions1);
     expect(spySendMail).toBeCalledWith(mailOptions2);
-    expect(spyClose).toBeCalledTimes(1);
+    expect(spyCloseTransport).toBeCalledTimes(1);
+  });
+
+  it('[OnPrem] should create transport/socket, send mails and close the transport/socket', async () => {
+    const spyCreateSocket = jest
+      .spyOn(SocksClient, 'createConnection')
+      .mockReturnValue(mockSocket as any);
+    const spyCreateTransport = jest
+      .spyOn(nodemailer, 'createTransport')
+      .mockReturnValue(mockTransport as any);
+    const spySendMail = jest.spyOn(mockTransport, 'sendMail');
+    const spyCloseTransport = jest.spyOn(mockTransport, 'close');
+    const spyEndSocket = jest.spyOn(mockSocket.socket, 'end');
+    const spyDestroySocket = jest.spyOn(mockSocket.socket, 'destroy');
+    const destination: any = {
+      originalProperties: {
+        'mail.password': 'password',
+        'mail.user': 'user',
+        'mail.smtp.host': 'smtp.gmail.com',
+        'mail.smtp.port': '587'
+      },
+      name: 'my-destination',
+      type: 'MAIL',
+      authentication: 'BasicAuthentication',
+      proxyType: 'OnPremise',
+      proxyConfiguration: {
+        host: 'smtp.gmail.com',
+        port: 587
+      }
+    };
+    const mailOptions1: MailOptions = {
```

Comment:
[q] Why is it called `mailOptions1`?
Is there also some other `mailOptions2`?
[req] Either way, if there is `mailOptions2` it should probably have a more specific name.